ratis-server/src/main/java/org/apache/ratis/server/impl/RaftServerImpl.java

```diff
@@ -963,6 +963,15 @@ void finishTransferLeadership() {
     }
   }
 
+  public RaftClientReply snapshotManually(SnapshotManuallyRequest request) throws IOException {
+    return waitForReply(request, snapshotManuallyAsync(request));
+  }
+
+  public CompletableFuture<RaftClientReply> snapshotManuallyAsync(SnapshotManuallyRequest request) throws IOException {
+    //TODO(codings-dan) : implement snapshot manually in server
```

Comment:
@codings-dan , we should implement the feature (i.e. this TODO) in the server first and then change the API (this pull request).

Reply:
Ok, I will try to implement this feature on the server side first